libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/MapboxTripStarter.kt

```diff
@@ -0,0 +1,214 @@
+        onTripDisabled(mapboxNavigation)
+        this.mapboxNavigation = null
+    }
+
+    /**
+     * Get the options that are currently set.
+     */
+    fun getOptions(): MapboxTripStarterOptions = optionsFlow.value
+
+    /**
+     * Set new options.
+     */
+    fun setOptions(options: MapboxTripStarterOptions) = apply {
+        this.optionsFlow.value = options
+    }
+
+    /**
+     * Get the location permission state. This may not reflect the actual permissions. Used for
+     * verifying the tests and will likely be removed.
+     */
+    @VisibleForTesting
+    internal fun getLocationPermissionGranted(): Boolean =
+        stateFlow.value.isLocationPermissionGranted
+
+    /**
+     * Set the location permission state. [MAPBOX_TRIP_STARTER_FOLLOW_DEVICE] will not work unless
+     * location permissions have been granted. If the location permissions are not granted and
+     * this is set to true, an error will be thrown.
+     */
+    @Throws(IllegalStateException::class)
+    fun setLocationPermissionGranted(granted: Boolean) {
+        stateFlow.value = stateFlow.value.copy(isLocationPermissionGranted = granted)
+    }
+
+    /**
+     * Enables a mode where the primary route is simulated by an artificial driver. Set the route
+     * with [MapboxNavigation.setNavigationRoutes].
+     *
+     * @param options optional options to use for route replay.
+     */
+    fun enableReplayRoute(
+        options: ReplayRouteSessionOptions? = null
+    ) = apply {
+        update { builder ->
+            options?.let { builder.replayRouteSessionOptions(options) }
+            builder.tripType(MAPBOX_TRIP_STARTER_REPLAY_ROUTE)
+        }
+    }
+
+    /**
+     * Throws an error if location permissions are set to true but the location permissions are
+     * not actually granted.
+     */
+    @Throws(IllegalStateException::class)
+    private fun checkState(state: MapboxTripStarterState) {
+        val mapboxNavigation = this.mapboxNavigation
+        checkNotNull(mapboxNavigation) {
+            "MapboxTripStarter cannot be used while MapboxNavigation is detached."
+        }
+        val context = mapboxNavigation.navigationOptions.applicationContext
+        val granted = state.isLocationPermissionGranted
+        if (granted && !PermissionsManager.areLocationPermissionsGranted(context)) {
+            error(
+                "updateLocationPermissions can only be set to true when location permissions" +
+                    " are granted."
+            )
+        }
+    }
+
+    // Directly calling this function from tests because the coroutine scope catches the exception
+    // https://github.com/Kotlin/kotlinx.coroutines/issues/1205
+    @VisibleForTesting
+    @Throws(IllegalStateException::class)
+    internal fun onStarterOptionsChanged(
+        mapboxNavigation: MapboxNavigation,
+        options: MapboxTripStarterOptions,
+        state: MapboxTripStarterState,
+    ) {
+        checkState(state)
```

Comment:
```suggestion
        checkState(mapboxNavigation, state)
```
This also lets you remove `mapboxNavigation` property in the class.

Reply:
Removing the checkState exception throwing in favor of a log warning